src/plugins/management/public/management_sections_service.ts

```diff
@@ -17,22 +17,47 @@
  * under the License.
  */
 
-import { ReactElement } from 'react';
 import { ManagementSection, RegisterManagementSectionArgs } from './utils';
-import { managementSections } from './components/management_sections';
+import {
+  IngestSection,
+  DataSection,
+  InsightsAndAlertingSection,
+  SecuritySection,
+  KibanaSection,
+  StackSection,
+} from './components/management_sections';
 
 import {
   ManagementSectionId,
   SectionsServiceSetup,
-  SectionsServiceStart,
   SectionsServiceStartDeps,
+  DefinedSections,
+  ManagementSectionsStartPrivate,
 } from './types';
+import { createGetterSetter } from '../../kibana_utils/public';
+
+const [getSectionsServiceStartPrivate, setSectionsServiceStartPrivate] = createGetterSetter<
+  ManagementSectionsStartPrivate
+>('SectionsServiceStartPrivate');
+
+export { getSectionsServiceStartPrivate };
 
 export class ManagementSectionsService {
-  private sections: Map<ManagementSectionId | string, ManagementSection> = new Map();
+  definedSections: DefinedSections;
 
-  private getSection = (sectionId: ManagementSectionId | string) =>
-    this.sections.get(sectionId) as ManagementSection;
+  constructor() {
+    // Note on adding sections - sections can be defined in a plugin and exported as a contract
```

Comment:
Can you provide a short example for what you mean?

Reply:
Here's a previous version of the code that was changed due to feedback - https://github.com/elastic/kibana/pull/71144/commits/291bf7a5656f7eecda4a265bce4fe54330bfcb32

Reply:
I mean an example in a comment